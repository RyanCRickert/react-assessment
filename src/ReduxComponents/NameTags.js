import React from 'react';
import { connect } from "react-redux";

import NameTag from "./NameTag";

export const NameTags = props => {
  return (
    <div className="Name-tag_holder">
    {props.names && props.names.split("\n").map((name, index) => {
      if(name.trim().length > 0) {
        return(
          <NameTag name={name} language={props.language} key={index} />
        )
      }
    })}
    </div>
  )
};

const mapStateToProps = (state) => {
  return {
    names: state.names.names,
    language: state.options.greeting
  };
}

export default connect(mapStateToProps)(NameTags);