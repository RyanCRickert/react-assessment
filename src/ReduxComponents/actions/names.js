export const changeNames = (names = "") => ({
  type: "SET_NAMES",
  names
})