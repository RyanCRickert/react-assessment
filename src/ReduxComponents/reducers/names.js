const namesReducerDefaultState = {
  names: ""
}

export default (state = namesReducerDefaultState, action) => {
  switch(action.type) {
    case "SET_NAMES":
      return {
        ...state,
        names: action.names
      }
    default:
      return state;
  }
}