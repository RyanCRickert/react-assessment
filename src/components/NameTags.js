import React from 'react';
import { AppConsumer } from "./AppProvider";

import NameTag from "./NameTag";

export default class NameTags extends React.Component {
  render() {
    return (
      <AppConsumer>
        {(context) => (
          <div className="Name-tag_holder">
          {context.state.namesArray && context.state.namesArray.map((person, index) => {
            if(person.trim().length > 0) {
              return( <NameTag name={person} language={context.state.language} color={context.state.color} changeColor={context.changeColor} key={index} /> )
            }
          })}
          </div>
        )}
    </AppConsumer>
    );
  }
}