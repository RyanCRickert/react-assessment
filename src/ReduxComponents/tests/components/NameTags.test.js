import React from "react";
import { shallow } from "enzyme";
import { NameTags } from "../../NameTags";
import names from "../fixtures/names";

test("should render NameTags with names", () => {
  const wrapper = shallow(<NameTags names={names} />)
  expect(wrapper).toMatchSnapshot();
});