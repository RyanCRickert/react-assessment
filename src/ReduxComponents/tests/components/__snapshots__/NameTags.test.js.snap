// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NameTags with names 1`] = `
<div
  className="Name-tag_holder"
>
  <NameTag
    key="0"
    name="Ray"
  />
  <NameTag
    key="1"
    name="Jay"
  />
  <NameTag
    key="2"
    name="Kay"
  />
  <NameTag
    key="3"
    name="May"
  />
  <NameTag
    key="4"
    name="Fay"
  />
</div>
`;
